pkg/apis/resource/validation/validation.go

```diff
@@ -331,6 +336,9 @@ func validateDeviceRequestAllocationResult(result resource.DeviceRequestAllocati
 	allErrs = append(allErrs, validateDriverName(result.Driver, fldPath.Child("driver"))...)
 	allErrs = append(allErrs, validatePoolName(result.Pool, fldPath.Child("pool"))...)
 	allErrs = append(allErrs, validateDeviceName(result.Device, fldPath.Child("device"))...)
+	if result.AdminAccess == nil && utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
```

Comment:
As noted elsewhere, you can't have really required fields behind gates.  One of the rules of API compatibility is that clients which used to work still work.  If the main API goes beta, clients will use that without this field.  Those clients can't be expected to magically learn about this field, unless there is a default value (e.g. defaults.go) which I do not see in this PR.

Reply:
Removed.

This was meant to ensure that if the feature gate is on, clients which have not been updated or have the feature disabled don't allocate a claim "incorrectly", but I see how that doesn't follow normal practices and also affects claims not using the feature.